ucore/sanoid.spec

```diff
@@ -0,0 +1,141 @@
+%global version 2.2.0
+%global git_tag v%{version}
+
+# Enable with systemctl "enable sanoid.timer"
+%global _with_systemd 1
+
+Name:		   sanoid
+Version:	   %{version}
+Release:	   1%{?dist}
+BuildArch:	   noarch
+Summary:	   A policy-driven snapshot management tool for ZFS file systems
+Group:		   Applications/System
+License:	   GPLv3
+URL:		   https://github.com/jimsalterjrs/sanoid
+Source0:	   https://github.com/jimsalterjrs/%{name}/archive/%{git_tag}/%{name}-%{version}.tar.gz
+
+Requires:	   perl-interpreter, mbuffer, lzop, pv, perl-Config-IniFiles, perl-Capture-Tiny, perl-Sys-Hostname
+%if 0%{?_with_systemd}
+Requires:      systemd >= 212
+
+BuildRequires: systemd
+%endif
+
+%description
+Sanoid is a policy-driven snapshot management
+tool for ZFS file systems. You can use Sanoid
+to create, automatically thin, and monitor snapshots
+and pool health from a single eminently
+human-readable TOML configuration file.
+
+%prep
+%setup -q
+
+%build
+echo "Nothing to build"
+
+%install
+%{__install} -D -m 0644 sanoid.defaults.conf %{buildroot}/etc/sanoid/sanoid.defaults.conf
+%{__install} -d %{buildroot}%{_sbindir}
+%{__install} -m 0755 sanoid syncoid findoid sleepymutex %{buildroot}%{_sbindir}
+
+%if 0%{?_with_systemd}
+%{__install} -d %{buildroot}%{_unitdir}
+%endif
+
+%if 0%{?fedora}
+%{__install} -D -m 0644 sanoid.conf %{buildroot}%{_docdir}/%{name}/examples/sanoid.conf
+%endif
+%if 0%{?rhel}
+%{__install} -D -m 0644 sanoid.conf %{buildroot}%{_docdir}/%{name}-%{version}/examples/sanoid.conf
+%endif
+
+%if 0%{?_with_systemd}
+cat > %{buildroot}%{_unitdir}/%{name}.service <<EOF
+[Unit]
+Description=Snapshot ZFS Pool
+Requires=zfs.target
+After=zfs.target
+
+[Service]
+Environment=TZ=UTC
+Type=oneshot
+ExecStart=%{_sbindir}/sanoid --cron
+EOF
+
+cat > %{buildroot}%{_unitdir}/%{name}.timer <<EOF
+[Unit]
+Description=Run Sanoid Every Minute
+
+[Timer]
+OnCalendar=*:0/1
+Persistent=true
+
+[Install]
+WantedBy=timers.target
+EOF
+
+%else
+%if 0%{?fedora}
+%{__install} -D -m 0644 sanoid.conf %{buildroot}%{_docdir}/%{name}/examples/sanoid.conf
+%endif
+%if 0%{?rhel}
+echo "* * * * * root %{_sbindir}/sanoid --cron" > %{buildroot}%{_docdir}/%{name}-%{version}/examples/sanoid.cron
+%endif
+%endif
+
+%post
+%{?_with_systemd:%{_bindir}/systemctl daemon-reload}
+
+%postun
+%{?_with_systemd:%{_bindir}/systemctl daemon-reload}
+
+%files
+%doc CHANGELIST VERSION README.md FREEBSD.readme
+%license LICENSE
+%{_sbindir}/sanoid
+%{_sbindir}/syncoid
+%{_sbindir}/findoid
+%{_sbindir}/sleepymutex
+%dir %{_sysconfdir}/%{name}
+%config %{_sysconfdir}/%{name}/sanoid.defaults.conf
+%if 0%{?fedora}
+%{_docdir}/%{name}
+%endif
+%if 0%{?rhel}
+%{_docdir}/%{name}-%{version}
+%endif
+%if 0%{?_with_systemd}
+%{_unitdir}/%{name}.service
+%{_unitdir}/%{name}.timer
+%endif
+
+%changelog
+* Mon Mar 18 2024 John McGee <john@johnmcgee.net> - 3.0.0
```

Comment:
> I'm not sure what the version would be since we don't want to get out of sync with the upstream release, but, would be good to reference our deps changes in the changelog.

I made this comment earlier in the PR review, and I admit it was vague.

We can't arbitrarily change the version of the package (eg, from 2.x to 3.x) because this version needs to reflect the version of the upstream software being installed.

Instead of `3.0.0` I suggest using `2.2.0-ucore1` for now, and we can increment the `ucoreN` in the future, if we need for subsequent tweaks specific to our packaging. 

Then, when upstream updates to `2.3.0` we'd package version `2.3.0-ucore1` but wouldn't need a changelog entry for it since our changes are already covered.


So that change needs to happen on this line, but see next comment (above in file) for where to put RPM version.